code/data-pipeline/src/data_pipeline/loader.py

```diff
@@ -303,9 +298,16 @@ def load_ilinet(self,
     return dat
 
 
-  def load_hhs(self, rates=True, drop_pandemic_seasons=True):
+  def load_hhs(self, rates=True, drop_pandemic_seasons=True, as_of=None):
     if drop_pandemic_seasons:
-      file_path = 'influenza-hhs/hhs.csv'
+      if as_of is None:
+        file_path = 'influenza-hhs/hhs.csv'
+      else:
+        # find the largest stored file dated on or before the as_of date
+        as_of_file_path = f'influenza-hhs/hhs-{str(as_of)}.csv'
+        all_file_paths = sorted(glob.glob('influenza-hhs/hhs-????-??-??.csv', root_dir = self.data_raw))
```

Comment:
It looks like the filenames in `data-raw` are under our control, so this note isn't a "requested change," just a suggestion based on past baggage of dealing with unannounced upstream file name changes.

This logic for getting the correct filename based on multiple variables is a good candidate to break out into its own supporting function (e.g., `get_file_path`), which can then be easily unit tested.

Reply:
I filed a new issue to keep track of this suggestion: #14